src/main/java/net/quedex/api/user/UserStream.java

```diff
@@ -90,30 +90,46 @@
     void batch(List<? extends OrderSpec> batch);
 
     /**
-     * Returns an object (not thread-safe) which may be used fluently to send a scheduled batch of {@link OrderSpec}s to the
-     * exchange. This batch will be executed between executionStartTimestamp and executionExpirationTimestamp.
-     * Calling {@link Batch#send()} sends scheduled batch of {@link OrderSpec}s to the exchange. In reaction to this
-     * the exchange engine creates a timer with the same id.
+     * <p>Returns an object (not thread-safe) which may be used fluently to create a time triggered batch of {@link OrderSpec}s.
+     * Calling {@link Batch#send()} sends this batch of {@link OrderSpec}s to the exchange.
+     * When a time triggered batch is received by the exchange engine, a new timer is registered.
+     * Based on the timer configuration, at some point in the future (between <i>executionStartTimestamp</i> and <i>executionExpirationTimestamp</i>),
+     * all the carried order commands are processed, one by one, in the creation order. If <i>executionStartTimestamp</i> is in the past
+     * and <i>executionExpirationTimestamp</i> is in the future, the batch will be processed immediately.</p>
+     * <p>Timers are ordered by <i>executionStartTimestamp</i> and order of arrival. Timers with earlier <i>executionStartTimestamp</i> are processed first.
+     * When there are more timers with the same <i>executionStartTimestamp</i> they are triggered by the order of arrival.</p>
+     * <p><strong>Note:</strong> timers are evaluated each time a new command is processed by the exchange engine. If between <i>executionStartTimestamp</i>
```

Comment:
Unnecessary implementation detail - remove. IMO from user perspective the important thing is granularity of execution and why `executionExpirationTimestamp` is relevant (and present at all).

Reply:
Removed. Details moved to native API documentation. Added a link to the native API link.